src/Package/Provider/BasicHttp/BasicHttpPackageProvider.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Meteor\Package\Provider\BasicHttp;
+
+use GuzzleHttp\Client;
+use Meteor\IO\IOInterface;
+use Meteor\Package\Provider\Exception\PackageNotFoundException;
+use Meteor\Package\Provider\PackageProviderInterface;
+
+class BasicHttpPackageProvider implements PackageProviderInterface
+{
+
+    /**
+     * @var IOInterface
+     */
+    private $io;
+
+    /**
+     * @var array
+     */
+    private $packageBaseUrls;
+
+    /**
+     * @var Client
+     */
+    private $httpClient;
+
+    /**
+     * BasicHttpPackageProvider constructor.
+     *
+     * @param IOInterface $io
+     * @param array $packageBaseUrls
+     * @param Client $httpClient
+     */
+    public function __construct(IOInterface $io, array $packageBaseUrls, Client $httpClient)
+    {
+        $this->io = $io;
+        $this->packageBaseUrls = $packageBaseUrls;
+        $this->httpClient = $httpClient;
+    }
+
+    /**
+     * @param string $packageName
+     * @param string $version
+     * @param string $tempDir
+     * @return string
+     * @throws PackageNotFoundException
+     */
+    public function download($packageName, $version, $tempDir)
+    {
+        if (!isset($this->packageBaseUrls[$packageName])) {
+            throw new PackageNotFoundException(sprintf('The HTTP base url for the "%s" package has not been configured.', $packageName));
+        }
+        $file = $version . '.zip';
+        $baseUrl = $this->packageBaseUrls[$packageName];
+
+        $packageURL = $baseUrl . $file;
+        try {
+            $response = $this->httpClient->request('GET', $packageURL, ['sink' => $tempDir . '/' . $file]);
+            if ($response->getStatusCode() == 200 && file_exists($tempDir . '/' . $file)) {
+                return $tempDir . '/' . $file;
```

Comment:
Do we need to remove the file after downloading it?

Reply:
Nope